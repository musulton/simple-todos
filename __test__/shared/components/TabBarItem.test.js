import {render, screen, fireEvent} from "@testing-library/react-native";

import TabBarItem from "../../../src/shared/components/TabBarItem";

describe("TabBarItem", () => {
    test("should render correctly when border is true", () => {
        const mockProps = {
            border: true,
            title: "title",
            setType: jest.fn(),
            type: "All",
            selected: false
        }

        render(<TabBarItem {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should render correctly when selected is true", () => {
        const mockProps = {
            border: false,
            title: "title",
            setType: jest.fn(),
            type: "All",
            selected: true
        }

        render(<TabBarItem {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should render correctly when title and type is equal", () => {
        const mockProps = {
            border: false,
            title: "ALl",
            setType: jest.fn(),
            type: "All",
            selected: false
        }

        render(<TabBarItem {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should call set type props with title props as params", () => {
        const mockProps = {
            border: true,
            title: "title",
            setType: jest.fn(),
            type: "title",
            selected: false
        }

        render(<TabBarItem {...mockProps} />)
        fireEvent.press(screen.getByText(mockProps.title))

        expect(mockProps.setType).toHaveBeenCalledWith(mockProps.title)
    })
})
