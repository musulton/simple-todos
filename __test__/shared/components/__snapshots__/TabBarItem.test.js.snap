// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarItem should render correctly 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#ffffff",
      },
      {
        "borderLeftColor": "#dddddd",
        "borderLeftWidth": 1,
      },
      {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#777777",
          "fontSize": 16,
        },
        {
          "fontWeight": "bold",
        },
      ]
    }
  >
    title
  </Text>
</View>
`;

exports[`TabBarItem should render correctly when border is true 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      null,
      {
        "borderLeftColor": "#dddddd",
        "borderLeftWidth": 1,
      },
      null,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#777777",
          "fontSize": 16,
        },
        null,
      ]
    }
  >
    title
  </Text>
</View>
`;

exports[`TabBarItem should render correctly when selected is true 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#ffffff",
      },
      null,
      null,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#777777",
          "fontSize": 16,
        },
        null,
      ]
    }
  >
    title
  </Text>
</View>
`;

exports[`TabBarItem should render correctly when title and type is equal 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      null,
      null,
      null,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#777777",
          "fontSize": 16,
        },
        null,
      ]
    }
  >
    ALl
  </Text>
</View>
`;
